pkg/audit/manager.go

```diff
@@ -280,17 +259,28 @@ func (am *Manager) audit(ctx context.Context) error {
 	timestamp := startTime.UTC().Format(time.RFC3339)
 	am.log = log.WithValues(logging.AuditID, timestamp)
 	logStart(am.log)
+	exportErrorMap := make(map[string]error)
+	if err := am.exportSystem.Publish(context.Background(), *auditConnection, *auditChannel, exportutil.ExportMsg{Message: "audit is started", ID: timestamp}); err != nil {
+		exportErrorMap[err.Error()] = err
```

Comment:
This can lead to memory consumption if by any chance driver returns unique error for all violations instead of returning one. Instead, we can modify returned error type to [ExportError](https://github.com/open-policy-agent/gatekeeper/pull/3832/files#diff-7dc8f318af1df0c3e2371e95dfa496cdb2762195e354ae132b99bbfa97ee5a89R25) and use error type as keys to limit the number of errors we are storing in map. Do folks think this is a way to go about limiting the number of errors stored? I am open to other suggestions.

I will make the change if we agree to use `ExportError` because that may need updating interface.

Reply:
agreed, your suggestion make sense. if we want to avoid interface changes, another idea might be to extract error category by checking with regex or before `:` for example

Reply:
Parsing the error message and using the string before `:` as key to store errors.